button-merchant/src/test/java/com/usebutton/merchant/ConnectionManagerTest.java

```diff
@@ -151,14 +171,17 @@ public MockResponse dispatch(RecordedRequest request) {
     }
 
     @Test(expected = ButtonNetworkException.class)
-    public void post_brokenResponse_catchException() throws Exception {
+    public void executeRequest_invalidResponse_shouldThrowError() throws Exception {
         server.setDispatcher(new Dispatcher() {
             @Override
             public MockResponse dispatch(RecordedRequest request) {
                 return new MockResponse().setBody("}{");
             }
         });
 
-        connectionManager.post("/", null);
+        connectionManager.executeRequest(new ApiRequest.Builder(ApiRequest.RequestMethod.POST,
+                "/")
+                .build()
```

Comment:
This seems like an unfortunate line wrap..

Reply:
Lol I wish...it's over the 100 character limit :(